golang/internal/health/server.go

```diff
@@ -0,0 +1,81 @@
+package health
+
+import (
+	"context"
+	"encoding/json"
+	"net"
+	"os"
+
+	"github.com/rs/zerolog/log"
+)
+
+var health = Status{
+	Connected: false,
+}
+
+func sendHealthData(conn net.Conn, healthData *Status) error {
+	data, err := json.Marshal(healthData)
+	if err != nil {
+		return err
+	}
+
+	_, err = conn.Write(data)
+	return err
+}
+
+func acceptLoop(socket net.Listener) {
+	for {
+		conn, err := socket.Accept()
+		if err != nil {
+			log.Error().Err(err).Msg("Health accept error")
+			break
+		}
+
+		err = sendHealthData(conn, &health)
+		if err != nil {
+			log.Error().Err(err).Msg("Failed to write health socket")
+		}
+
+		err = conn.Close()
+		if err != nil {
+			log.Error().Err(err).Msg("Failed to close health socket")
+		}
+	}
+}
+
+func SetHealthGRPCStatus(connected bool) {
+	health.Connected = connected
+}
+
+func Serve(ctx context.Context) error {
+	_, err := os.Stat(socketPath)
+	if err == nil {
+		err = os.RemoveAll(socketPath)
```

Comment:
The first error is swallowed and only the remove error is printed. Is it intentional/ fail-safe?